pennylane/_qubit_device.py

```diff
@@ -270,18 +270,17 @@ def statistics(self, observables):
                 results.append(self.probability(wires=obs.wires))
 
             elif obs.return_type is State:
-                if len(obs.wires) != self.num_wires:
-                    raise QuantumFunctionError("The state must be returned over all wires")
+                if len(observables) > 1:
+                    raise QuantumFunctionError(
+                        "The state cannot be returned in combination with other return types"
+                    )
 
-                try:
-                    state = self.state
-                except AttributeError:
-                    state = None
+                if self.wires.labels != tuple(range(self.num_wires)):
+                    raise QuantumFunctionError(
+                        "Returning the state is not supported when using " "custom wire labels"
+                    )
```

Comment:
@trbromley and I were discussing and decided there are too many possible ambiguities here. This solution gives us the possibility to reach the vast majority of users while deferring unlikely but difficult edge cases to be further supported later on